picard/ui/logview.py

```diff
@@ -37,7 +41,11 @@ def __init__(self, title, logger, w=740, h=340, parent=None):
         self.browser.setDocument(self.doc)
         self.vbox = QtWidgets.QVBoxLayout(self)
         self.vbox.addWidget(self.browser)
-        self._display()
+        self._setup_formats()
+        self.verbosity = self.default_verbosity
+        self.hidden_domains = set()
```

Comment:
Again, don't get why both domains and show only domains are needed? Does selecting domains discard messages altogether and `show only` create a temp filter? If so, we should keep the distinction between filters and records separate.

Reply:
Because domains is a tuple (even if a trick allows to pass just a string, when the tuple should have only one element), example:

- a message A with domains `('ws', 'ratecontrol')`
- a message B with domains `('ws')`
- a message C with domains `('ratecontrol')`
- a message D with domains `('ws', 'ratecontrol', 'subratecontrol')`

Show only `(ws)` shows A,B,D (only)
Show only `(ws, ratecontrol)` shows A, B, C,D

Show only `(ws, ratecontrol)` + hide `('subratecrontrol')` shows A, B, C but not D

Show only totally unset + hide ('ws') shows C only

That said, i didn't write tests to ensure the actual behaviour is the one i expect ;)

Do you get why both are needed ? Include list + exclude list, both working together to provide more than you could do with just one.